src/app/lab/components/units-dropdown.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Input } from '@angular/core';
+
+import { Chart } from '../../models/chart.model';
+
+/*  Units Dropdown Component
+
+    -- Requires project input
+    Expected use:
+        <ccl-units-dropdown
+            [chart]="your_chart">
+*/
+
+@Component({
+  selector: 'ccl-units-dropdown',
+  template: `<div dropdown class="dropdown dropdown-units">
+              <button dropdownToggle class="button dropdown-toggle" type="button"
+                id="unitsDropdown" data-toggle="dropdown" aria-haspopup="true"
+                aria-expanded="true">
+                <i class="icon-flash"></i>
+                {{ chart.unit }}
+                <i class="icon-angle-down caret"></i>
+              </button>
+              <ul *dropdownMenu class="dropdown-menu" aria-labelledby="unitsDropdown">
+                <li *ngFor="let unit of chart.indicator.available_units">
+                  <a (click)="onUnitSelected(unit)"
+                    placement="bottom">{{ unit }}</a>
+                </li>
+              </ul>
+            </div>`
+})
+export class UnitsDropdownComponent {
+
+    @Input() chart: Chart;
+
+    constructor() {}
```

Comment:
I poked at the component section of the [Angular style guide](https://angular.io/guide/styleguide#components) and didn't see a preference explicitly mentioned, but every example omits the constructor when blank. I would generally agree -- if it's not required that it be implemented then it should be omitted if its not doing anything.

This is probably a more meta discussion, but in general we should defer to the Angular style guide for consistency across projects.